plugins/quickinspector/quickitemgeometry.cpp

```diff
@@ -112,6 +142,10 @@ void QuickItemGeometry::initFrom(QQuickItem *item)
         QQuickItemPrivate *parentPriv = QQuickItemPrivate::get(parent);
         parentTransform = parentPriv->itemToWindowTransform();
     }
+
+    traceColor = colorForItem(item);
+    traceTypeName = ObjectDataProvider::typeName(item);
+    traceName = ObjectDataProvider::name(item);
```

Comment:
Yeah the ObjectDataProvider was the only required thing from the gammaray_core.
What do you mean by moving this to the caller ?